presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -98,6 +99,8 @@ public ExchangeNode(
         checkArgument(scope != REMOTE || type == Type.REPARTITION || !partitioningScheme.isReplicateNullsAndAny(), "Only REPARTITION can replicate remotely");
 
         orderingScheme.ifPresent(ordering -> {
+            PartitioningHandle partitioningHandle = partitioningScheme.getPartitioning().getHandle();
+            checkArgument(scope != REMOTE || partitioningHandle.equals(SINGLE_DISTRIBUTION), "remote merging exchange requires single distribution");
```

Comment:
This change belongs to `Add ordering to RemoteSourceNode and ExchangeNode` commit.